Add NullaWay to package-info for o.s.i.config package

cppwfs opened the pull request:
- Add `@org.jspecify.annotations.NullAway` to package-info.java
- Apply `@Nullable`, `@SuppressWarnings("NullAway.Init)` as need. Tweak code as necessary in locations where NullAway can not discern that the null condition is already mitigated

cppwfs committed:
Add NullaWay to package-info for o.s.i.config package

Add @org.jspecify.annotations.NullAway to package-info.java
Apply @Nullable, @SuppressWarnings("NullAway.Init) as need.
Tweak code as necessary in locations where NullAway can not descern that the null condition is already mitigated

artembilan commented:
Pay attention how those annotations are converted GH user handles:
<img width="694" height="99" alt="image" src="https://github.com/user-attachments/assets/76450e26-aefb-4883-b6e9-0297273e1f19" />
That's why I prefer to use back-ticks even in commit messages for code snippets to avoid calling those GH users for nothing 😄

cppwfs commented:
Thank you for catching that!    Updated the comment.

artembilan commented:
>  Updated the comment.

It doesn't help any more.
You have already invited those GH users to our PR review 😉 

I just mean the reasoning behind my preference to use code snippet tools in commit messages as well.

artembilan commented on spring-integration-core/src/main/java/org/springframework/integration/config/AbstractSimpleMessageHandlerFactoryBean.java:
```diff
@@ -312,7 +317,7 @@ public boolean isSingleton() {
 	}
 
-	private static Object extractTarget(Object object) {
+	private static @Nullable Object extractTarget(@Nullable Object object) {
```
I think this method has to be changed to be non-nullable.
And we should have something like in the body:
```
return Objects.requireNonNullElse(AopProxyUtils.getSingletonTarget(object), object);
```

I doubt that we need recursion here.
But I might be wrong if some tests would fail.
Forth checking anyway!

artembilan commented on spring-integration-core/src/main/java/org/springframework/integration/config/CorrelationStrategyFactoryBean.java:
```diff
@@ -39,8 +39,10 @@
 	private Object target;
 
+	@SuppressWarnings("NullAway.Init")
 	private String methodName;
```
This can be null according to the class logic.
While for the `target` we need to ensure that it is not-null.

artembilan commented on spring-integration-core/src/main/java/org/springframework/integration/config/DefaultConfiguringBeanFactoryPostProcessor.java:
```diff
@@ -79,6 +81,7 @@ public class DefaultConfiguringBeanFactoryPostProcessor implements BeanDefinitio
 	private static final Set<Integer> REGISTRIES_PROCESSED = new HashSet<>();
 
+	@Nullable
 	private static final Class<?> XPATH_CLASS;
```
Please, put the `@Nullable` just before the `Class<?>`.
While it works with JSpecify, but better to follow its code style recommendation.

artembilan commented on spring-integration-core/src/main/java/org/springframework/integration/config/SpelFunctionFactoryBean.java:
```diff
@@ -42,9 +44,9 @@ public class SpelFunctionFactoryBean implements FactoryBean<Method>, Initializin
+	private @Nullable String functionName;
 
-	private Method method;
+	private @Nullable Method method;
```
The `@SuppressWarnings("NullAway.Init")` on both properties.
One of them is from `setBeanName()` and another from the `afterPropertiesSet()`.
I doubt we should pursue the case when this class is not used as a bean in the code.

artembilan reviewed (changes requested):
Very cool!
Thanks

cppwfs committed:
Checkpoint

cppwfs committed:
Remove Objects.requireNonNull surround this.handler in the AbstractSimpleMessageHandlerFactoryBean

Remove commented code in AbstractSimpleMessageHandlerFactoryBean.extractTarget
Restructure IntegrationManagementConfigurer.registerComponentGauges so that a NullAway suppresion can be removed from the method
related to: https://github.com/spring-projects/spring-integration/issues/10083
rebased
